geoengine/colorizer.py

```diff
@@ -1,39 +1,36 @@
 """This module is used to generate geoengine compatible color map definitions as a json string."""
 
 import json
-from ast import Tuple
+from typing import Tuple
 import numpy as np
 from matplotlib.colors import ListedColormap
 from matplotlib.cm import ScalarMappable
 
 
 class Colorizer():
     """This class is used to generate geoengine compatible color map definitions as a json string."""
-
-    def __init__(self, steps: int = 10, min_max: Tuple(int, int) = (0, 255)):
+    # pylint: disable=too-few-public-methods
+
+    def __init__(
+            self,
+            steps: int = 10,
+            min_max: Tuple[int, int] = None,
+            default_color: Tuple[int, int, int, int] = (0, 0, 0, 0),
+            no_data_color: Tuple[int, int, int, int] = (0, 0, 0, 0)):
```

Comment:
Please add the `map_name` as the first param here.

Then, please set min_max as second parameter without default value.

Then, you can have steps, default and nodata with default values.